src/scripts/collapsed_queue.js

```diff
@@ -0,0 +1,57 @@
+import { getPostElements } from '../util/interface.js';
+import { exposeTimelines } from '../util/react_props.js';
+import { getPreferences } from '../util/preferences.js';
+import { onNewPosts } from '../util/mutations.js';
+import { keyToCss } from '../util/css_map.js';
+
+const excludeClass = 'xkit-collapsed-queue-done';
+const wrapperClass = 'xkit-collapsed-queue-wrapper';
+const containerClass = 'xkit-collapsed-queue-container';
+
+let timelineRegex;
+let footerSelector;
+
+const processPosts = async function () {
+  await exposeTimelines();
+
+  getPostElements({ excludeClass, timeline: timelineRegex }).forEach(async postElement => {
+    const headerElement = postElement.querySelector('header');
+    const footerElement = postElement.querySelector(footerSelector);
+
+    const wrapper = Object.assign(document.createElement('div'), { className: wrapperClass });
+    const container = Object.assign(document.createElement('div'), { className: containerClass });
+    wrapper.append(container);
+
+    headerElement.after(wrapper);
+    while (wrapper.nextElementSibling !== footerElement) {
+      container.append(wrapper.nextElementSibling);
+    }
+  });
+};
+
+export const main = async function () {
+  const { runInQueue, runInDrafts } = await getPreferences('collapsed_queue');
+  const regexGroup = [
+    ...runInQueue ? ['queue'] : [],
+    ...runInDrafts ? ['draft'] : []
+  ].join('|');
+  timelineRegex = new RegExp(String.raw`\/v2\/blog\/[^/]+\/posts\/(${regexGroup})`);
```

Comment:
Since we're using the `RegExp` constructor rather than literal notation, these forward-slashes don't need to be escaped. They have no special meaning in regular expressions themselves.

```suggestion
  timelineRegex = new RegExp(String.raw`/v2/blog/[^/]+/posts/(${regexGroup})`);
```

Reply:
Oh, man, I'm still not used to wrapping my brain around the difference between "what you have to type to create a regex" and "the characters that make up the regex" :D The `String.raw` call should be unnecessary as well, then, if I'm understanding correctly.